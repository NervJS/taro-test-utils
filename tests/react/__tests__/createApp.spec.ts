import TestUtils from '@tarojs/test-utils-react'
import { navigateTo } from '@tarojs/router'

const wait = (time: number) => new Promise(resolve => setTimeout(() => { resolve(null) }, time))

describe('CreateApp', () => {

  let testUtils: TestUtils

  beforeAll(async () => {
    testUtils = new TestUtils()
    await testUtils.createApp()
  })

  test('app lifecycle', async () => {
    await testUtils.AppLifecycle.onLaunch()
    expect(testUtils.queries.queryByAttribute('launch', '1')).toBeTruthy()
    await testUtils.AppLifecycle.onShow()
    expect(testUtils.queries.queryByAttribute('show', '1')).toBeTruthy()
  })

  test('page lifecycle', async () => {
    await testUtils.PageLifecycle.onShow('pages/index/index')
    await wait(10)
    expect(testUtils.html()).toMatchSnapshot()
  })

  test('router change', async () => {
    navigateTo({
      url: '/pages/second/index'
    })
    await testUtils.PageLifecycle.onShow('pages/second/index')
    await wait(10)
    expect(testUtils.queries.waitForQueryByText('seoncd page show')).toBeTruthy()
  })


  test('trigger lifecycle', async () => {
    await testUtils.PageLifecycle.triggerReachBottom('pages/second/index')
    await wait(10)
    expect(testUtils.queries.waitForQueryByText('seoncd page reachBottom')).toBeTruthy()
  })
})