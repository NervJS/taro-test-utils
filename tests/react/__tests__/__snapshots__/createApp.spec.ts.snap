// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateApp page lifecycle 1`] = `
"[36m<body[39m
  [33mlaunch[39m=[32m"1"[39m
  [33mshow[39m=[32m"1"[39m
[36m>[39m
  [36m<div[39m
    [33mclass[39m=[32m"taro_router"[39m
    [33mid[39m=[32m"app"[39m
  [36m>[39m
    [36m<div[39m
      [33mclass[39m=[32m"taro_page taro_page_show taro_page_stationed"[39m
      [33mid[39m=[32m"/pages/index/index?stamp=AA"[39m
    [36m>[39m
      [36m<taro-view-core[39m
        [33mclass[39m=[32m"index"[39m
      [36m>[39m
        [36m<taro-text-core[39m
          [33mclass[39m=[32m""[39m
        [36m>[39m
          [0mFirst Page[0m
        [36m</taro-text-core>[39m
        [36m<taro-text-core[39m
          [33mclass[39m=[32m""[39m
        [36m>[39m
          [0mfirst page load[0m
        [36m</taro-text-core>[39m
        [36m<taro-text-core[39m
          [33mclass[39m=[32m""[39m
        [36m>[39m
          [0mfirst page show[0m
        [36m</taro-text-core>[39m
        [36m<taro-text-core[39m
          [33mclass[39m=[32m""[39m
        [36m/>[39m
      [36m</taro-view-core>[39m
    [36m</div>[39m
  [36m</div>[39m
[36m</body>[39m"
`;
